src/services/user/user.class.js

```diff
@@ -43,11 +43,44 @@ class Service {
       };
     });
 
+    let level = 1;
+
+    if(xp && xp.length) {
+      const values = rules['levels']['values'];
+      const levelXP = xp.find(x => x['name'] === 'XP')['amount'];
```

Comment:
I find this code a bit hard to read at the moment; I think naming this variable `currentXP` would help a lot.